modular_skyrat/modules/ashwalkers/code/buildings/ash_tendril.dm

```diff
@@ -0,0 +1,164 @@
+#define REQUIRED_OBSERVERS 2
+#define MEGAFAUNA_MEAT_AMOUNT 20
+
+//this is for revitalizing/preserving regen cores
+/obj/structure/lavaland/ash_walker/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(!istype(attacking_item, /obj/item/organ/internal/regenerative_core))
+		return ..()
+
+	if(!user.mind.has_antag_datum(/datum/antagonist/ashwalker))
+		balloon_alert(user, "must be an ashwalker!")
+		return
+
+	var/obj/item/organ/internal/regenerative_core/regen_core = attacking_item
+	regen_core.preserved()
+	playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+	balloon_alert_to_viewers("[src] revitalizes [regen_core]!")
+	return
+
+//this is for logging the destruction of the tendril
+/obj/structure/lavaland/ash_walker/Destroy()
+	var/compiled_string = "The [src] has been destroyed at [loc_name(src.loc)], nearest mobs are "
+	var/found_anyone = FALSE
+
+	for(var/mob/living/carbon/carbons_nearby in range(7))
+		compiled_string += "[key_name(carbons_nearby)],"
+		found_anyone = TRUE
+
+	if(!found_anyone)
+		compiled_string += "nobody."
+
+	log_game(compiled_string)
+	return ..()
+
+//this is for transforming a person into an ashwalker
+/obj/structure/lavaland/ash_walker/attack_hand(mob/living/user, list/modifiers)
+	if(!ishuman(user))
+		return
+
+	var/mob/living/carbon/human/human_user = user
+	if(istype(human_user.dna.species, /datum/species/lizard/ashwalker))
+		return
+
+	var/allow_transform = 0
+
+	for(var/mob/living/carbon/human/count_human in range(2, src))
+		if(!istype(count_human.dna.species, /datum/species/lizard/ashwalker))
```

Comment:
Don't we have like `isashwalker()` as a helper macro already?

Reply:
we do not.